### ISSUE TITLE
test

### ISSUE THREAD

**Issue by ml-and-ml** [nderus/DeepHit-PyTorch#1]
are there any test results to verify integrity of the pytorch implementation?

**Comment by nderus** [comment]
Full comparison with the published results in the paper is a bit tricky because of a couple reasons. Mainly,  in the original implementation performance is evaluated as a grid of time-specific concordance index over some time horizon T, while in the paper it  is reported as a single value (e.g., 0.69 for METABRIC). I might have to review the supplementary to check if they specify whether the single value is obtained via integration (and if so, how) or by looking at the latest time horizon T. In the latter case we obtain a c-index of around 0.66 in METABRIC.

Because the original is in TF and my version in PyTorch, there may be minor differences, moreover the original implementation provides  a seed for the train-test split but not for weight initialisation.

**Comment by ml-and-ml** [comment]
I don't think the validation necessarily needs to be 1 to 1 with paper. I'm just asking about tests regarding implementation of loss to make sure the loss is doing what it should be doing. Not sure if you have applied this model successfully in other applications. Thanks btw for doing the work to translate this to pytorch

**Comment by nderus** [comment]
I see what you mean, from what I can tell, the code does what is supposed to.

In terms of proof of concept I can say that  in METABRIC the hyper parameters search for the rank loss weight (alpha) seem to converge consistently around the same value (0.5), suggesting that i) is not a random solution ii) the model with higher values of alpha outperforms the one with lower values. This should be reproducible using the code

We are using these losses  as one of the baselines in a work that is currently under review. Results seem to be in line with the literature (e.g., DeepHit performances in https://openreview.net/pdf?id=1mU6ADbjk-c)
